File: LEETCODE/3. Longest Substring Without Repeating Characters/Solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        String ans="";
        String st="";
        
        
        for(int i=0;i<s.length();i++){
            char ch = s.charAt(i);
            
            
            int ind = st.indexOf(ch);
            //System.out.println(ind);
            
            if(ind==-1){
                st = st+ch;
            }else{
                if(st.length()>ans.length()){
                    ans=st;
                    System.out.println(ans);
                }
                st = st.substring(ind+1)+ch;
            }
            
        }
        
        if(st.length()>ans.length())
                ans=st;
        
        return ans.length();
    }
}

File: LEETCODE/3. Longest Substring Without Repeating Characters/Solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = ''
        st=''
        
        for x in s:
            if x in st:
                z=st.index(x)
                if len(st)>len(ans):
                    ans=st
                st=st[z+1:]+x
            else:
                st=st+x
        
        if len(st)>len(ans):
            ans=st
            
        return len(ans)
        